system_monitor/src/system_monitor/monitor.py

```diff
@@ -39,27 +41,68 @@ def validate_single_param(param_name, required_type):
     rospy.init_node('system_monitor', anonymous=True)
     if validate_params():
         pub = rospy.Publisher('/system_workload', WorkloadMsg, latch=True, queue_size=1)
+        diagnostic_pub = rospy.Publisher('/diagnostics', DiagnosticArray, latch=True, queue_size=1)
 
         hostname = socket.gethostname()
         config = rospy.get_param('/system_monitor')
+        rate = rospy.Rate(config['update_frequency'])
 
-        while not rospy.is_shutdown():
-            start_time = rospy.Time.now()
+        diag_array = DiagnosticArray()
+        diag_cpu = DiagnosticStatus()
+        # start all names with "SYSTEM" for diagnostic analysesr
+        diag_cpu.name = "SYSTEMCPU"
+        diag_cpu.hardware_id = "CPU"
+        diag_mem = DiagnosticStatus()
+        diag_mem.name = "SYSTEMMemory"
+        diag_cpu.hardware_id = "Memory"
 
-            running_processes, cpu_usages = cpus.collect_all() if config['do_cpu'] else (-1, [])
+        while not rospy.is_shutdown():
+            running_processes, cpu_usages, overall_usage_percentage = cpus.collect_all() if config['do_cpu'] else (
+                -1, [])
```

Comment:
```suggestion
                -1, [], 0)
```
Or some other default for `overall_usage_percentage`